From 424a50d452c306ec780d7e9596b306afdf716ab0 Mon Sep 17 00:00:00 2001
From: Uri Mashiach <uri.mashiach@compulab.co.il>
Date: Wed, 29 Nov 2017 09:08:53 +0200
Subject: [PATCH 34/57] arm: imx7d: cl-som-imx7: update initalization groups

The board initialization APIs are called by global initialization APIs.
Update the board APIs division to the global APIs:
* Move layout setup APIs from board_init() to the checkboard() global
  API.
* Move cl_som_imx7_setup_i2c1() from board_late_init() to the
  checkboard() global API.
* Move cl_som_imx7_get_baseboard_id() from board_init() to the
  checkboard() global API.
* Move cl_som_imx7_update_dtb_name() from board_late_init() to
  the checkboard() global API.
* Move cl_som_imx7_set_serial_env() from board_late_init() to
  the checkboard() global API.
* Move the environment variable board_name set from board_late_init() to
  the checkboard() global API.

Signed-off-by: Uri Mashiach <uri.mashiach@compulab.co.il>
---
 board/compulab/cl-som-imx7/cl-som-imx7.c | 45 ++++++++++++------------
 configs/cl-som-imx7_defconfig            |  1 +
 include/configs/cl-som-imx7.h            |  3 ++
 3 files changed, 26 insertions(+), 23 deletions(-)

diff --git a/board/compulab/cl-som-imx7/cl-som-imx7.c b/board/compulab/cl-som-imx7/cl-som-imx7.c
index b4ee39dae6..d72aff4a39 100644
--- a/board/compulab/cl-som-imx7/cl-som-imx7.c
+++ b/board/compulab/cl-som-imx7/cl-som-imx7.c
@@ -583,30 +583,12 @@ int board_early_init_f(void)
 
 int board_init(void)
 {
-	int ret;
-
 	/* address of boot parameters */
 	gd->bd->bi_boot_params = PHYS_SDRAM + 0x100;
 	cl_som_imx7_setup_i2c0();
 	cl_som_imx7_setup_gpmi_nand();
 	cl_som_imx7_setup_fec();
 	cl_som_imx7_spi_init();
-	ret = cl_eeprom_layout_setup(&cl_som_am57x_layout,
-				     cl_som_am57x_eeprom_buf,
-				     LAYOUT_VERSION_AUTODETECT,
-				     CONFIG_SYS_I2C_EEPROM_BUS,
-				     CONFIG_SYS_I2C_EEPROM_ADDR);
-	if (cl_som_imx7_base_i2c_init)
-		ret |= cl_eeprom_layout_setup(&sb_som_am57x_layout,
-					      sb_som_am57x_eeprom_buf,
-					      LAYOUT_VERSION_AUTODETECT,
-					      CL_SOM_IMX7_I2C_BUS_EXT,
-					      CL_SOM_IMX7_I2C_EEPROM_EXT);
-
-       if (ret)
-               printf("EEPROM layout initialization failure\n");
-
-       cl_som_imx7_get_baseboard_id();
 
 	return 0;
 }
@@ -660,24 +642,20 @@ int board_late_init(void)
 {
 	int ret;
 
-	cl_som_imx7_base_i2c_init = cl_som_imx7_setup_i2c1();
-
 #ifdef CONFIG_VIDEO_MXS
 	ret = enable_display();
 	if (ret < 0)
 		printf("Display enable failure\n");
 #endif /* CONFIG_VIDEO_MXS */
 
-	setenv("board_name", "CL-SOM-iMX7");
 	cl_som_imx7_setup_wdog();
-	cl_som_imx7_update_dtb_name();
-	cl_som_imx7_set_serial_env();
 
 	return 0;
 }
 
 int checkboard(void)
 {
+	int ret;
 	char *mode;
 
 	if (IS_ENABLED(CONFIG_ARMV7_BOOT_SEC_DEFAULT))
@@ -687,6 +665,27 @@ int checkboard(void)
 
 	printf("Board: CL-SOM-iMX7 in %s mode\n", mode);
 
+	setenv("board_name", "CL-SOM-iMX7");
+	cl_som_imx7_base_i2c_init = cl_som_imx7_setup_i2c1();
+	ret = cl_eeprom_layout_setup(&cl_som_am57x_layout,
+				     cl_som_am57x_eeprom_buf,
+				     LAYOUT_VERSION_AUTODETECT,
+				     CONFIG_SYS_I2C_EEPROM_BUS,
+				     CONFIG_SYS_I2C_EEPROM_ADDR);
+	if (ret)
+		printf("Module EEPROM layout initialization failure\n");
+	if (cl_som_imx7_base_i2c_init)
+		ret = cl_eeprom_layout_setup(&sb_som_am57x_layout,
+					     sb_som_am57x_eeprom_buf,
+					     LAYOUT_VERSION_AUTODETECT,
+					     CL_SOM_IMX7_I2C_BUS_EXT,
+					     CL_SOM_IMX7_I2C_EEPROM_EXT);
+	if (ret)
+		printf("Base board EEPROM layout initialization failure\n");
+	cl_som_imx7_get_baseboard_id();
+	cl_som_imx7_update_dtb_name();
+	cl_som_imx7_set_serial_env();
+
 	return 0;
 }
 
diff --git a/configs/cl-som-imx7_defconfig b/configs/cl-som-imx7_defconfig
index 7937766d6b..a6eb2cf9a3 100644
--- a/configs/cl-som-imx7_defconfig
+++ b/configs/cl-som-imx7_defconfig
@@ -18,6 +18,7 @@ CONFIG_OF_BOARD_SETUP=y
 CONFIG_SYS_EXTRA_OPTIONS="IMX_CONFIG=arch/arm/imx-common/spl_sd.cfg"
 CONFIG_SPI_BOOT=y
 CONFIG_BOOTDELAY=3
+# CONFIG_DISPLAY_BOARDINFO is not set
 CONFIG_SPL=y
 CONFIG_SPL_BOARD_INIT=y
 CONFIG_SYS_MMCSD_RAW_MODE_U_BOOT_USE_SECTOR=y
diff --git a/include/configs/cl-som-imx7.h b/include/configs/cl-som-imx7.h
index 620dbf5a7e..ba0ab8cecd 100644
--- a/include/configs/cl-som-imx7.h
+++ b/include/configs/cl-som-imx7.h
@@ -248,4 +248,7 @@
 #define CONFIG_ENV_CALLBACK_LIST_STATIC "displaytype:displaytype"
 #endif /* CONFIG_VIDEO */
 
+/* Miscellaneous */
+#define CONFIG_DISPLAY_BOARDINFO_LATE
+
 #endif	/* __CONFIG_H */
-- 
2.17.1

